bundles/framework/myplaces3/view/PlaceForm.js

```diff
@@ -92,104 +85,132 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.view.PlaceForm',
-                    placeLink = placeLink.replace('>', '');
-                }
-                var imageLink = onScreenForm.find('input[data-name=imagelink]').val(),
-                    categorySelection = onScreenForm.find('select[data-name=category]').val();
-                if (typeof categorySelection === 'string') {
-                    categorySelection = parseInt(categorySelection);
-                }
-                values.place = {
-                    name: placeName,
-                    desc: placeDesc,
-                    attention_text: placeAttention,
-                    link: placeLink,
-                    imageLink: imageLink,
-                    category: forcedCategory || categorySelection
-                };
-                if (this.placeId) {
-                    values.place.id = this.placeId;
+            const form = this._getOnScreenForm();
+            if (form.length === 0) {
+                return;
+            }
+            var forcedCategory = (this.options ? this.options.category : undefined);
+            const place = this.place || Oskari.clazz.create('Oskari.mapframework.bundle.myplaces3.model.MyPlace');
+            const errors = [];
+            const name = form.find('input[data-name=placename]').val();
+            const desc = form.find('input[data-name=placedesc]').val();
+            const att = form.find('input[data-name=placeAttention]').val();
+            // Validate values
+            if (!name) {
+                errors.push({
+                    name: 'name',
+                    error: this.loc('validation.placeName')
+                });
+            } else if (Oskari.util.sanitize(name) !== name) {
+                errors.push({
+                    name: 'name',
+                    error: this.loc('validation.placeNameIllegal')
+                });
+            }
+            if (Oskari.util.sanitize(desc) !== desc) {
+                errors.push({
+                    name: 'desc',
+                    error: this.loc('validation.descIllegal')
+                });
+            }
+            /*
+            TODO: Should we validate attention_text. Localization is missing!
+            if (Oskari.util.sanitize(att) !== att) {
+                errors.push({
+                    name: 'att',
+                    error: this.loc('validation.attIllegal')
+                });
+            }
+            */
+            if (errors.length > 0) {
+                return { errors };
+            }
+            place.setName(name);
+            place.setDescription(desc);
+            place.setAttentionText(att);
+            let placeLink = form.find('input[data-name=placelink]').val();
+            if (placeLink) {
+                if (placeLink.indexOf('://') === -1 || placeLink.indexOf('://') > 6) {
+                    placeLink = 'http://' + placeLink;
                 }
+                placeLink = placeLink.replace('<', '');
+                placeLink = placeLink.replace('>', '');
```

Comment:
This look weird. What is this for?

Reply:
It was there and change comes from formatting. Removed. If there is some weird need for this then it could be returned.